chain/client/src/chunk_validation.rs

```diff
@@ -440,6 +401,101 @@ fn pre_validate_chunk_state_witness(
     })
 }
 
+/// Validate that receipt proofs contain the receipts that should be applied during the
+/// transition proven by ChunkStateWitness. The receipts are extracted from the proofs
+/// and arranged in the order in which they should be applied during the transition.
+/// TODO(resharding): Handle resharding properly. If the receipts were sent from before
+/// a resharding boundary, we should first validate the proof using the pre-resharding
+/// target_shard_id and then extract the receipts that are targeted at this half of a split shard.
+fn validate_source_receipt_proofs(
+    source_receipt_proofs: &HashMap<ChunkHash, ReceiptProof>,
+    blocks_after_last_last_chunk: &[Block],
+    last_chunk_block: &Block,
+    target_chunk_shard_id: ShardId,
+) -> Result<Vec<Receipt>, Error> {
+    let mut receipts_to_apply = Vec::new();
+    let mut expected_proofs_len = 0;
+
+    // Iterate over blocks between last_chunk_block (inclusive) and last_last_chunk_block (exclusive),
+    // from the newest blocks to the oldest.
+    let blocks_after_last_last_without_last_last = blocks_after_last_last_chunk
```

Comment:
use [split_last](https://doc.rust-lang.org/std/primitive.slice.html#method.split_last)

Reply:
This is no longer necessary after e2f74a074f5bc9a3d3f1930afb7927c799447e9a